src/algorithms/peps_opt.jl

```diff
@@ -155,8 +164,8 @@ function fixedpoint(
             end
             return costfun(ψ, envs´, H)
         end
-        # withgradient returns tuple of gradients `g`
-        return E, only(g)
+        g = only(g)  # withgradient returns tuple of gradients `g`
+        return E, g
```

Comment:
```suggestion
        return E, only(g)  # withgradient returns tuple of gradients `g`
```
While it might be a little more readable, you should be careful with re-using the same variable name for things like this. Sometimes, the compiler gets confused and can no longer figure out the type of `g`, especially in contexts like these with anonymous functions etc.
So I would either keep it as it was, or use `E, gs` for the return values of `withgradient` and then `g = only(gs)`.
I don't have a preference about that :)